### PR TITLE
TST/CLN: deduplicate troublesome rank values

### PR DESCRIPTION
**Pull request by mzeitlin11**
xref https://github.com/pandas-dev/pandas/pull/38681#discussion_r548554225

### CODE REVIEW

File: pandas/conftest.py [hunk 1]
```diff
@@ -591,6 +593,91 @@ def narrow_series(request):
     return _narrow_series[request.param].copy()
 
 
+_dtype_nuisance_arr_map = {
```
Comment (jreback) [inline pandas/conftest.py:596]: can you add a comment here about where this is used

File: pandas/conftest.py [hunk 2]
```diff
@@ -591,6 +593,91 @@ def narrow_series(request):
...
+        1,
+        9999,
+        100000,
+        1e10,
+        np.iinfo(np.int64).max,
+    ],
+    "object": [NegInfinity(), "1", "A", "BA", "Ba", "C", Infinity()],
+}
+
+_dtype_na_map = {
```
Comment (jreback) [inline pandas/conftest.py:641]: can you put this *inside* the fixture, this is generally a special case

File: pandas/conftest.py (hunk 1 above)
Comment (jreback) [inline pandas/conftest.py:596]: put rank in the name

**Review by jreback — CHANGES_REQUESTED** [review]
this is ok for now. i think if we keep adding more shared fixtures (good thing), this file might bloat, so maybe have to do something later.

cc @jbrockmendel

File: pandas/conftest.py (hunk 1 above)
Comment (mzeitlin11) [reply pandas/conftest.py]: Done

File: pandas/conftest.py (hunk 2 above)
Comment (mzeitlin11) [reply pandas/conftest.py]: Done

File: pandas/conftest.py (hunk 1 above)
Comment (mzeitlin11) [reply pandas/conftest.py]: Done

File: pandas/tests/series/methods/test_rank.py [hunk 3]
```diff
@@ -206,90 +205,10 @@ def test_rank_signature(self):
...
-            values = np.insert(values, nan_indices, na_value)
-            exp_order = np.insert(exp_order, nan_indices, np.nan)
-        # shuffle the testing array and expected results in the same way
-        random_order = np.random.permutation(len(values))
-        iseries = Series(values[random_order])
-        exp = Series(exp_order[random_order], dtype="float64")
-        iranks = iseries.rank()
-        tm.assert_series_equal(iranks, exp)
+    def test_rank_inf(self, nuisance_rank_series_and_expected):
+        series, expected = nuisance_rank_series_and_expected
```
Comment (jreback) [inline pandas/tests/series/methods/test_rank.py:209]: actually an alternative here is simply to use `frame_or_series` fixture and leave the fixture in tests/frame/methods (e.g. this is what we do for things for both series & frame), i think in this case it makes more sense.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/methods/test_rank.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/series/methods/test_rank.py:209]: yah i think this makes a lot more sense than a relatively-complicated fixture in conftest

Comment (mzeitlin11) [reply pandas/tests/series/methods/test_rank.py]: Thanks that is much nicer, good pattern to know

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
perfect, ping on green

**Comment by mzeitlin11** [comment]
green

**Comment by jreback** [comment]
thanks

**Pull request merged**